addons/website_sale/controllers/main.py

```diff
@@ -576,16 +576,16 @@ def checkout_form_validate(self, mode, all_form_values, data):
         error = dict()
         error_message = []
 
-        if data.get('partner_id'):
+        if mode[1] == 'billing' and data.get('partner_id'):
             partner_su = request.env['res.partner'].sudo().browse(int(data['partner_id'])).exists()
-            name_change = partner_su and 'name' in data and data['name'] != partner_su.name
-            email_change = partner_su and 'email' in data and data['email'] != partner_su.email
+            name_change = partner_su and partner_su.name and 'name' in data and (data['name'] or False) != (partner_su.name or False)
+            email_change = partner_su and partner_su.email and 'email' in data and (data['email'] or False) != (partner_su.email or False)
```

Comment:
I don't believe that this change is good:
- It is correct to consider that there is a name change when the name was unset and a new value is provided.
- Both left parts of the `or` are already formatted as `False` when falsy.

Reply:
yes, I think we actually only need the `partner_su.name` part, I will adjust it :pray:

Reply:
:face_with_spiral_eyes: 

```py
In [4]: p = env.user.partner_id.create({'name': 'ok', 'email':False})

In [5]: p.email
Out[5]: False

In [6]: p.email != ''
Out[6]: True
```

not tested, but probably not useless...
As already explained before your previously wrong commit:
https://github.com/odoo/odoo/pull/111708#discussion_r1236730422

Reply:
In practice, [this does not happen](https://github.com/odoo/odoo/blob/14.0/addons/website_sale/controllers/main.py#L669-L670).

Reply:
I remember @paju-odoo had a special case about a portal user with an empty name, but I forgot how to reproduce the exact example. But that case should be blocked by the added  `and partner_su.name` part now. @paju-odoo could you tell me how did you make that test case? if you still remember :pray:

Reply:
@AntoineVDV :ok: never see this patch before... A fix of 6 months ago in v14 :face_in_clouds:  In this case, :+1: 

ty for feedback

Reply:
Hi @Jinjiu96,
The use case in the following: 

- Create a partner with main address
- Create a delivery address associated with your partner, leave the name empty
- Try to purchase anything from this user's account
- The name is empty and when you fill it in, you get an error message telling you that you cannot change it ("invoices have already been issued").

@AntoineVDV is aware of this use case, we tested it together last week.